sites/public/src/components/listings/search/ListingsSearchForm.tsx

```diff
@@ -0,0 +1,163 @@
+import React, { useEffect, useState } from "react"
+import { ButtonGroup } from "./ButtonGroup"
+import { MultiSelectGroup } from "./MultiSelectGroup"
+import { ListingSearchParams, parseSearchString } from "../../../lib/listings/search"
+
+type ListingsSearchFormProps = {
+  searchString?: string
+  bedrooms: FormOption[]
+  bathrooms: FormOption[]
+  counties: FormOption[]
+  onSubmit: (params: ListingSearchParams) => void
+}
+
+export type FormOption = {
+  label: string
+  value: string
+  isDisabled?: boolean
+}
+
+const containerStyle: React.CSSProperties = {
+  width: "600px",
+  margin: "15px auto",
+  border: "1px solid black",
+}
+
+const headerStyle: React.CSSProperties = {
+  padding: "3px 15px",
+  borderBottom: "1px solid black",
+}
+
+const inputSectionStyle: React.CSSProperties = {
+  margin: "0px 15px",
+  borderTop: "1px solid black",
+}
+
+const textInputStyle: React.CSSProperties = {
+  border: "1px solid black",
+  padding: "2px 4px",
+  margin: "5px",
+}
+
+const footerStyle: React.CSSProperties = {
+  display: "flex",
+  padding: "3px 15px",
+  borderTop: "1px solid black",
+}
+
+const clearButtonStyle: React.CSSProperties = {
+  textDecoration: "underline",
+}
+
+/**
+ * A form for searching listings
+ *
+ * @param props
+ * @returns
+ */
+export function ListingsSearchForm(props: ListingsSearchFormProps) {
+  const searchString = props.searchString || ""
+
+  const nullState: ListingSearchParams = {
+    bedrooms: null,
+    bathrooms: null,
+    monthlyRent: "",
+    // Using an empty array here is how we know to look for an array value when parsing
+    counties: [],
+  }
+
+  const initialState = parseSearchString(searchString, nullState)
```

Comment:
This enables us to define a specific set of search parameters at the beginning.  When the Alameda frontend is deprecated, for instance, it could redirect to our search page with `?search=counties:Alameda` and it would automatically show only listings that are only in Alameda County.  This is also how a search from the landing page could show results on the listings page.  

While we could use a named parameter for each value ( ie `?counties[]=Alameda&bathrooms=2`), this approach seemed more future-proof, easier to inject, and offers a bit of abstraction and validation around the expected values.  I didn't see much in the way of downsides other than having to write a parser/builder, but that was fairly easy to do and is already in place.  I'm curious if there might be a solid advantage to doing it another way that I hadn't considered, though